root/templates/recipe/edit.tt

```diff
@@ -13,7 +13,7 @@
 </div>
 [% END %]
 
-<form method="post" action="[% update_url %]">
+<form class="confirmIfUnsavedChanges" method="post" action="[% update_url %]">
```

Comment:
I checked out your branch and tested. The JS file was loaded but it had no effect. I made changes and could leave the page without warning.